spring-cloud-gcp-pubsub-stream-binder/src/main/java/com/google/cloud/spring/stream/binder/pubsub/PubSubMessageChannelBinder.java

```diff
@@ -75,8 +78,15 @@ protected MessageHandler createProducerMessageHandler(
       ExtendedProducerProperties<PubSubProducerProperties> producerProperties,
       MessageChannel errorChannel) {
 
-    PubSubMessageHandler messageHandler =
-        new PubSubMessageHandler(this.pubSubTemplate, destination.getName());
+    PubSubMessageHandler messageHandler = new PubSubMessageHandler(this.pubSubTemplate, destination.getName());
+
+    HeaderMapper<Map<String, String>> headerMapper = new PubSubHeaderMapper();
+
+    if (producerProperties.getExtension().getAllowedHeaders() != null) {
```

Comment:
Consider extracting `producerProperties.getExtension().getAllowedHeaders()` into a local variable; also check if it can be null.